bep032tools/generator/tests/test_BEP032Generator.py

```diff
@@ -184,7 +184,56 @@ def test_generate_example_structure(self):
     def doCleanups(self):
         initialize_test_directory(clean=True)
 
+class Test_FolderGeneration(unittest.TestCase):
+
+    def setUp(self):
+        self.test_dir = Path(initialize_test_directory(clean=True))
+        os.mkdir(self.test_dir / 'generateTest')
+        csv_filename = generate_simple_csv_file()
+        self.csv_file = csv_filename
+
+    def test_generate_folder(self):
+        """
+        When the file did not exist, the generation did not work properly.
+        Now if the folder does not exist, the function creates the folder before the generation function is launched
+        Parameters
+        ----------
+        Returns
+        ----------
+        bool
+            True if the generation is well done. False if it failed.
+        """
+        self.test_dir = self.test_dir / 'test'
+        test_generate = False
+        if os.path.isdir(self.test_dir):
+            BEP032Data.generate_struct(self.csv_file, self.test_dir)
+            test_generate = True
+        else:
```

Comment:
this else condition will never be run, as the `self.test_dir` folder is always generated by the `setUp` method before running the test.